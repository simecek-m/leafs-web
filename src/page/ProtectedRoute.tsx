import { useAuth0 } from "@auth0/auth0-react";
import { HOME_PATH } from "constant";
import { Navigate, Outlet } from "react-router-dom";

const ProtectedRoute = () => {
  const { isLoading, error, isAuthenticated } = useAuth0();
  if (isLoading) return <div>loading</div>;
  if (error) return <div>Oooooops, {error.message}</div>;
  if (isAuthenticated) {
    return <Outlet />;
  } else {
    return <Navigate replace to={HOME_PATH} />;
  }
};

export default ProtectedRoute;
