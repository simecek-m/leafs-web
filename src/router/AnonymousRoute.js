import React from "react";
import { Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";

export function AnonymousRoute({ component, user, ...rest }) {
  return user ? (
    <Redirect to="/lists" />
  ) : (
    <Route {...rest} component={component} />
  );
}

const mapStateToProps = state => {
  return {
    user: state.user
  };
};

export default connect(mapStateToProps)(AnonymousRoute);
