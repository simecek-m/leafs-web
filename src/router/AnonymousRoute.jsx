import { useAuth0 } from "@auth0/auth0-react";
import LoadingPage from "page/LoadingPage";
import React from "react";
import { Navigate, Outlet } from "react-router-dom";

export default function AnonymousRoute() {
  const { user, isLoading } = useAuth0();
  if (isLoading) {
    return <LoadingPage />;
  }
  if (user) {
    return <Navigate to="/list" replace />;
  }
  return <Outlet />;
}
