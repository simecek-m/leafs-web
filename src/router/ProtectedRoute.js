import React from "react";
import { Route, Redirect } from "react-router-dom";

export default function ProtectedRoute({ component, ...rest }) {
  // TODO: inject authentication
  const user = null;

  return user ? <Route {...rest} component={component} /> : <Redirect to="/" />;
}
