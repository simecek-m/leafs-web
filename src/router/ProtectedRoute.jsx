import { useAuth0 } from "@auth0/auth0-react";
import LoadingPage from "page/LoadingPage";
import React from "react";
import { Navigate, Outlet } from "react-router-dom";

export default function ProtectedRoute() {
  const { user, isLoading } = useAuth0();
  if (isLoading) {
    return <LoadingPage />;
  }
  if (user) {
    return <Outlet />;
  }
  return <Navigate to="/" replace />;
}
