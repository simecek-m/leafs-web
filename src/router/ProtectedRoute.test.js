import React from "react";
import { ProtectedRoute } from "router/ProtectedRoute";
import { shallow } from "enzyme";

const TEST_MOCK_COMPONENT = () => <div>Component</div>;
const TEST_USER_JWT = "xxxxx.yyyyy.zzzzz";

describe("ProtectedRoute component", () => {
  test("should render Redirect component", () => {
    const wrapper = shallow(<ProtectedRoute user={null} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("should render Route to props component", () => {
    const wrapper = shallow(
      <ProtectedRoute user={TEST_USER_JWT} component={TEST_MOCK_COMPONENT} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
