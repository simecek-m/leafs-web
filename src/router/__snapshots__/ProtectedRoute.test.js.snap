// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute component should render Redirect component 1`] = `
<Redirect
  to="/login"
/>
`;

exports[`ProtectedRoute component should render Route to props component 1`] = `
<Route
  component={[Function]}
/>
`;
